simple/stats/stat_var_hierarchy_generator.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from stats.data import Triple
+import stats.stat_var_hierarchy_constants as svh_constants
+
+
+class StatVarHierarchyGenerator:
+  """Given a list of input triples (including stat vars), 
+generates a SV hierarchy and returns a list of output triples
+representing the hierarchy.
+"""
+
+  def __init__(self, triples: list[Triple]) -> None:
+    # Build SV PVs.
+    self.sv_pvs = StatVarPVs(triples)
+    # TODO: Create SVG + SV tree.
+    # TODO: Generate SVG + SV triples.
+
+
+class StatVarPVs:
+  """An intermediate helper object that extracts SVs from input triples
+  and puts their PVs in a dict for quick lookup later.
+  It also maintains a separate dict for population types since SV hierarchies
+  are rooted at a group representing their population type.
+  """
+
+  def __init__(self, triples: list[Triple]) -> None:
+    self.sv_id_2_population_type: dict[str, str] = {}
+    self.sv_id_2_pvs: dict[str, dict[str, str]] = {}
+
+    # Collect all triples into pop type and pv dictionaries.
+    # Also collect SV DCIDs for filtering subsequently.
+
+    # Using dict instead of set to maintain order.
```

Comment:
Dict doesn't maintain order unless you use OrderedDict I think?

Reply:
As of Python 3.7, they maintain insertion order: https://stackoverflow.com/a/39980744

Reply:
Ah, I see. And comment why order matters?

Reply:
Done.